mne/tests/test_source_space.py

```diff
@@ -66,17 +66,15 @@ def test_add_patch_info():
         s['pinfo'] = None
 
     # test that no patch info is added for small dist_limit
-    try:
-        add_source_space_distances(src_new, dist_limit=0.00001)
-    except RuntimeError:  # what we throw when scipy version is wrong
-        pass
-    else:
-        assert all(s['nearest'] is None for s in src_new)
-        assert all(s['nearest_dist'] is None for s in src_new)
-        assert all(s['pinfo'] is None for s in src_new)
+    add_source_space_distances(src_new, dist_limit=0.00001)
+    assert all(s['nearest'] is None for s in src_new)
+    assert all(s['nearest_dist'] is None for s in src_new)
+    assert all(s['pinfo'] is None for s in src_new)
 
-    # now let's use one that works
-    add_source_space_distances(src_new)
+    # now let's use one that works (and test our warning-throwing)
+    monkeypatch.setattr(mne.source_space, '_DIST_WARN_LIMIT', 1)
+    with pytest.warns(RuntimeWarning, match='Computing distances for 258'):
+        add_source_space_distances(src_new)
```

Comment:
I <3 `pytest` for stuff like this. I originally created a `contextmanager` that monkey-patched our code, then had a sense that `pytest` probably already provided a mechanism to do it, and it did. Hooray!